R/render.R

```diff
@@ -147,9 +147,24 @@ moon_reader = function(
     }
   }
 
+  hook_highlight_output = if (isTRUE(nature$highlightLines)) {
+    # an ugly way to access the `output` hook of markdown output in knitr
+    hook_output = local({
+      ohooks = knitr::knit_hooks$get(); on.exit(knitr::knit_hooks$restore(ohooks))
+      knitr::render_markdown()
+      knitr::knit_hooks$get('output')
+    })
```

Comment:
This is repeating the code in `hook_highlight()`. Are you able to avoid the repetition? You can return a list `knitr::knit_hooks$get(c('source', 'output'))` in `local()`.